test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/ConfigurationManagerTests.cs

```diff
@@ -462,22 +465,34 @@ public void ValidateOpenIdConnectConfigurationTests(ConfigurationManagerTheoryDa
         {
             TestUtilities.WriteHeader($"{this}.ValidateOpenIdConnectConfigurationTests");
             var context = new CompareContext();
+            OpenIdConnectConfiguration configuration;
+            var configurationManager = new ConfigurationManager<OpenIdConnectConfiguration>(theoryData.MetadataAddress, theoryData.ConfigurationRetreiver, theoryData.DocumentRetriever, theoryData.ConfigurationValidator);
+
+            if (theoryData.PresetCurrentConfiguration)
+                TestUtilities.SetField(configurationManager, "_currentConfiguration", new OpenIdConnectConfiguration() { Issuer = Default.Issuer });
 
             try
             {
                 //create a listener and enable it for logs
                 var listener = TestUtils.SampleListener.CreateLoggerListener(EventLevel.Warning);
 
-                var configurationManager = new ConfigurationManager<OpenIdConnectConfiguration>(theoryData.MetadataAddress, theoryData.ConfigurationRetreiver, theoryData.DocumentRetriever, theoryData.ConfigurationValidator);
-                var configuration = configurationManager.GetConfigurationAsync().Result;
+                configuration = configurationManager.GetConfigurationAsync().Result;
 
                 if (!string.IsNullOrEmpty(theoryData.ExpectedErrorMessage) && !listener.TraceBuffer.Contains(theoryData.ExpectedErrorMessage))
                     context.AddDiff($"Expected exception to contain: '{theoryData.ExpectedErrorMessage}'.{Environment.NewLine}Log is:{Environment.NewLine}'{listener.TraceBuffer}'");
 
+                theoryData.ExpectedException.ProcessNoException(context);
             }
-            catch (Exception ex)
+            catch (AggregateException ex)
             {
-                theoryData.ExpectedException.ProcessException(ex, context);
+                // this should throw, because last configuration retrieved was null
+                Assert.Throws<AggregateException>(() => configuration = configurationManager.GetConfigurationAsync().Result);
```

Comment:
>Assert [](http://example.com/codeflow?start=16&length=6)

Let's add a check here, if the theoryData.PresetCurrentConfiguration is set to true, throw since this should never be the case.